ext/opcache/jit/zend_jit_helpers.c

```diff
@@ -1965,7 +1965,12 @@ static zval* ZEND_FASTCALL zend_jit_fetch_obj_r_slow_ex(zend_object *zobj)
 	void **cache_slot = CACHE_ADDR(opline->extended_value & ~ZEND_FETCH_OBJ_FLAGS);
 
 	retval = zobj->handlers->read_property(zobj, name, BP_VAR_R, cache_slot, result);
-	if (retval == result && UNEXPECTED(Z_ISREF_P(retval))) {
+	if (retval != result) {
+		/* On exception, we may free the result in the ZEND_HANDLE_EXCEPTION VM handler
+		 * so we need to initialize the result.
+		 * We don't need to copy because we use the return value directly as a register. */
+		ZVAL_UNDEF(result);
+	} else if (UNEXPECTED(Z_ISREF_P(retval))) {
```

Comment:
This may cause problems for case without exceptions, because the type in memory is changed and it may be inconsistent with inferred type.

I think, it's better to use ``zend_jit_check_exception_undef_result()`` instead of ``zend_jit_check_exception()`` when ``Z_MODE(res_addr) == IS_REG``.